homeassistant/components/sensor/tof.py

```diff
@@ -0,0 +1,129 @@
+"""Platform for Time of Flight sensor VL53L1X."""
+
+import asyncio
+import logging
+from functools import partial
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components import rpi_gpio
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['VL53L1X2==0.1.5']
+
+DEPENDENCIES = ['rpi_gpio']
+
+_LOGGER = logging.getLogger(__name__)
+
+LENGTH_MILLIMETERS = 'mm'
+
+CONF_I2C_ADDRESS = 'i2c_address'
+CONF_I2C_BUS = 'i2c_bus'
+CONF_XSHUT = 'xshut'
+
+DEFAULT_NAME = 'VL53L1X'
+DEFAULT_I2C_ADDRESS = 0x29
+DEFAULT_I2C_BUS = 1
+DEFAULT_XSHUT = 16
+DEFAULT_RANGE = 2
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME,
+                 default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_I2C_ADDRESS,
+                 default=DEFAULT_I2C_ADDRESS): vol.Coerce(int),
+    vol.Optional(CONF_I2C_BUS,
+                 default=DEFAULT_I2C_BUS): vol.Coerce(int),
+    vol.Optional(CONF_XSHUT,
+                 default=DEFAULT_XSHUT): cv.positive_int,
+})
+
+
+def init_tof(xshut, level):
+    """XSHUT port LOW resets the device."""
+    rpi_gpio.setup_output(xshut)
+    rpi_gpio.write_output(xshut, level)
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_entities,
+                               discovery_info=None):
+    """Reset and initialize the VL53L1X ToF Sensor from STMicroelectronics."""
+    from VL53L1X2 import VL53L1X  # pylint: disable=import-error
+
+    name = config.get(CONF_NAME)
+    bus_number = config.get(CONF_I2C_BUS)
+    i2c_address = config.get(CONF_I2C_ADDRESS)
+    unit = LENGTH_MILLIMETERS
+    xshut = config.get(CONF_XSHUT)
+
+    await hass.async_add_executor_job(
+        init_tof, xshut, 0
+    )
+    await asyncio.sleep(0.01)
+    await hass.async_add_executor_job(
+        init_tof, xshut, 1
+    )
+    await asyncio.sleep(0.01)
+
+    sensor = await hass.async_add_executor_job(
+        partial(VL53L1X, bus_number)
+    )
+
+    sensor.open()
```

Comment:
Does this do I/O?

Reply:
Don´t seem to do I/I, just malloc() to initialize driver variables.